File: Data Structures & Algorithms/k-closest-points-to-origin/submission-0.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        vector<vector<int>> retArr;
        priority_queue<pair<double, vector<int>>, vector<pair<double, vector<int>>>, greater<>> heap;
        
        for (auto x : points) {
            double res = sqrt(pow(x[0], 2) + pow(x[1], 2));
            heap.push({res, x});
        }
        
        for (int i = 0; i < k; i++) {
            retArr.push_back(heap.top().second);
            heap.pop();
        }
        return retArr;
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = [] # pair {elem, index}
        retArr = [0] * len(temperatures)

        for i in range(len(temperatures)):
            while stack and temperatures[i] > stack[-1][0]:
                retArr[stack[-1][1]] = i - stack[-1][1]
                stack.pop()
            stack.append((temperatures[i], i))
        
        return retArr


        